sdk/workloads/Azure.ResourceManager.Workloads/tests/Tests/SviCRUDTests.cs

```diff
@@ -0,0 +1,235 @@
+            await sapDatabaseInstanceResource.GetTagResource().DeleteAsync(WaitUntil.Completed);
+
+            firsttag = sapCentralServerInstanceResource.Data.Tags.Values.GetEnumerator();
+            firsttag.MoveNext();
+            Assert.AreEqual(firsttag.Current, "TestValue1");
+            await sapCentralServerInstanceResource.GetTagResource().DeleteAsync(WaitUntil.Completed);
+
+            firsttag = sapApplicationServerInstanceResource.Data.Tags.Values.GetEnumerator();
+            firsttag.MoveNext();
+            Assert.AreEqual(firsttag.Current, "TestValue1");
+            await sapApplicationServerInstanceResource.GetTagResource().DeleteAsync(WaitUntil.Completed);
+        }
+
+        /// <summary>
+        /// Gets an object of SAP Workload Resource for Single Server that should be used for PUT call.
+        /// </summary>
+        /// <param name="infraRgName"> Contains the infra RG
+        /// used as app resource group. </param>
+        /// <returns>PHP Workload Resource.</returns>
+        private SapVirtualInstanceData GetSingleServerPayloadToPut(string infraRgName, bool isInstall)
+        {
+            string sshPrivateKey = File.ReadAllText(Path.Combine(
+                Path.GetDirectoryName(Assembly.GetExecutingAssembly().Location), "SshKeyPrivate.txt"));
+
+            string sshPublicKey = File.ReadAllText(Path.Combine(
+                Path.GetDirectoryName(Assembly.GetExecutingAssembly().Location), "SshKeyPublic.txt"));
+
+            if (isInstall)
+            {
+                var testSapWorkloadJson = File.ReadAllText(Path.Combine(
+                    Path.GetDirectoryName(Assembly.GetExecutingAssembly().Location), "SingleServerInstall.json"));
+
+                dynamic DynamicObject = JsonConvert.DeserializeObject<dynamic>(testSapWorkloadJson);
+
+                DynamicObject.properties.configuration.infrastructureConfiguration.appResourceGroup =
+                    infraRgName;
+                DynamicObject.properties.configuration.infrastructureConfiguration.
+                    virtualMachineConfiguration.osProfile.osConfiguration.ssh.publicKeys[0].keyData =
+                        sshPublicKey;
+                DynamicObject.properties.configuration.softwareConfiguration.sshPrivateKey =
+                    sshPrivateKey;
+
+                var sapPayloadJson = JsonDocument.Parse(JsonConvert.SerializeObject(DynamicObject)).RootElement;
+
+                SapVirtualInstanceData testSapWorkload = SapVirtualInstanceData.DeserializeSapVirtualInstanceData(
+                    sapPayloadJson);
+
+                return testSapWorkload;
+            }
+            else
+            {
+                var testSapWorkloadJson = File.ReadAllText(Path.Combine(
+                    Path.GetDirectoryName(Assembly.GetExecutingAssembly().Location), "SingleServerCreate.json"));
+
+                dynamic DynamicObject = JsonConvert.DeserializeObject<dynamic>(testSapWorkloadJson);
```

Comment:
repeated code?

Reply:
This code has to be executed on the create Json and Install Json differently, in the separate code flows.

Reply:
Changed it to your suggested code.